src/Estimators/InputSection.h

```diff
@@ -0,0 +1,116 @@
+
+  std::unordered_set<std::string> attributes;  // list of attribute variables
+  std::unordered_set<std::string> parameters;  // list of parameter variables
+  std::unordered_set<std::string> required;    // list of required variables
+
+  std::unordered_set<std::string> strings;     // list of string variables
+  std::unordered_set<std::string> bools;       // list of boolean variables
+  std::unordered_set<std::string> integers;    // list of integer variables
+  std::unordered_set<std::string> reals;       // list of real variables
+  
+  std::unordered_map<std::string, std::any> default_values; // default values for optional variables
+  
+private:
+  // Storage for variable values read from XML, etc.
+  std::unordered_map<std::string, std::any> values;
+  
+public:
+  // Query functions
+  bool is_attribute(std::string name){return attributes.find(name)!=attributes.end();};
+  bool is_parameter(std::string name){return parameters.find(name)!=parameters.end();};
+  bool is_required(std::string name){return required.find(name)!=required.end();};
+
+  bool is_string(std::string name){return strings.find(name)!=strings.end();};
+  bool is_bool(std::string name){return bools.find(name)!=bools.end();};
+  bool is_integer(std::string name){return integers.find(name)!=integers.end();};
+  bool is_real(std::string name){return reals.find(name)!=reals.end();};
+
+  bool has_default(std::string name){return default_values.find(name)!=default_values.end();};
+  bool has(std::string name){return values.find(name)!=values.end();};
+
+
+  // Enable read-only access to variable values.
+  //   Needs updating to allow copy-less return.
+  template<typename T>
+  T get(std::string name){return std::any_cast<T>(values[name]);};
+
+
+  // Later replace this with initializer_list/unordered_map initializer.
+  //   Enforce correctness checks and immutability/read-only access thereafter.
+  //template<typename T>
+  //void set(std::string name, T& value)
+  //{
+  //  const std::type_info& Ttype = typeid(T);
+  //  bool correct_type = (Ttype == typeid(std::string)) & is_string(name);
+  //  correct_type |= (Ttype == typeid(bool)) & is_bool(name);
+  //  correct_type |= (Ttype == typeid(int))  & is_integer(name);
+  //  correct_type |= (Ttype == typeid(Real)) & is_real(name);
+  //  if (!correct_type)
+  //    throw UniformCommunicateError("InputSection::set  incorrect type assignment attempted");
+  //  values[name] = value;
+  //};
+
+
+  // Read variable values (initialize) from XML input.
+  //   Later, this should call a correctness checking function and enforce immutability.
+  //   (required values are all provided, provided values fall in allowed ranges)
+  void readXML(xmlNodePtr cur);
+
+  //  Simple write of contents.  Can be replaced/removed in any final implemenation.
+  void report();
+
+private:
+  // Perform typed read and assignment of input variables
+  void set_from_stream(std::string name, std::istringstream& svalue);
```

Comment:
Usualy for name.
`const std::string& name` can be better than `std::string name`

Reply:
Done

Reply:
[std::string_view](https://en.cppreference.com/w/cpp/string/basic_string_view)?

Reply:
string_view is a C++17 feature. I think it makes sense in some use cases. In QMCPACK string operation is never the major computational cost, so leave contributors to decide whether to opt-in this feature.